### ISSUE TITLE
Zeebe Exporter shows a large number of unexported messages

### ISSUE THREAD

**Issue by 0xtej** [camunda/zeebe#11238]
**Description**

In [zeebe.json](https://github.com/camunda/zeebe/blob/main/monitor/grafana/zeebe.json) what is the autocommit interval defined for zeebe exporter?

We are using https://github.com/camunda-community-hub/zeebe-kafka-exporter and exporting only some events via kafka.  We are seeing a buildup of around 18k records not exported yet using for this [metric](https://github.com/camunda/zeebe/blob/6217d07ef93adfdffeb9426921cbdb66b94b6b75/monitor/grafana/zeebe.json#L1789) during off-traffic hours (night)

I suspect this is because we consuming messages of only one type PROCESS_INSTANCE. This is our zeebe-kafka-exporter config, we are ignoring a whole bunch of other event types. This build of 18k records is reducing after some time, leading me to suspect there is some autocommit interval, what is this value and is it configurable?

```# Controls which records are pushed to Kafka and to which topic
          # Each entry is a sub-map which can contain two entries:
          #     type => string
          #     topic => string
          #
          # Topic is the topic to which the record with the given value type
          # should be sent to, e.g. for a deployment record below we would
          # send the record to "zeebe-deployment" topic.
          #
          # Type is a comma separated string of accepted record types, allowing you to filter if you
          # want nothing (""), commands ("command"), events ("events"), or rejections ("rejection"),
          # or a combination of the three, e.g. "command,event".
          #
          # To omit certain records entirely, set type to an empty string. For example,
          # records:
          #   deployment: { type: "" }
          records:
            # If a record value type is omitted in your configuration file,
            # it will fall back to whatever is configured in the defaults
            defaults: 
              type: "event"
              topic: ${ZEEBE_KAFKA_TOPIC_NAME}
            # For records with a value of type DEPLOYMENT
            deployment: { type: "" }
            # For records with a value of type DEPLOYMENT_DISTRIBUTION
            deploymentDistribution: { type: "" }
            # For records with a value of type ERROR
            error: { type: "" }
            # For records with a value of type INCIDENT
            incident: { type: "" }
            # For records with a value of type JOB_BATCH
            jobBatch: { type: "" }
            # For records with a value of type JOB
            job: { type: "" }
            # For records with a value of type MESSAGE
            message: { type: "" }
            # For records with a value of type MESSAGE_SUBSCRIPTION
            messageSubscription: { type: "" }
            # For records with a value of type MESSAGE_START_EVENT_SUBSCRIPTION
            messageStartEventSubscription: { type: "" }
            # For records with a value of type PROCESS
            process: { type: "" }
            # For records with a value of type PROCESS_EVENT
            processEvent: { type: "" }
            # For records with a value of type PROCESS_INSTANCE
            # processInstance: { topic: zeebe-process-instance }
            # For records with a value of type PROCESS_INSTANCE_RESULT
            processInstanceResult: { type: "" }
            # For records with a value of type PROCESS_MESSAGE_SUBSCRIPTION
            processMessageSubscription: { type: "" }
            # For records with a value of type TIMER
            timer: { type: "" }
            # For records with a value of type VARIABLE
            variable: { type: "" }```

**Comment by korthout** [comment]
Sorry for the long wait @0xtej. The team will investigate your question and respond in the coming two weeks.

**Comment by remcowesterhoud** [comment]
Hi @0xtej, sorry for the late response.

First of all I think it's important to point out that these metrics are an approximation. The `zeebe_log_appender_last_committed_position` and the `zeebe_exporter_last_exported_position` are updated by prometheus asynchronously. It is supposed to give a general idea about the state of the system, but not a real-time overview.
If you want Prometheus to scrape this data more often it would involve increasing the `scrape_interval` of Prometheus. The default is 15 seconds, we have our own monitoring set to 5 seconds. We have also set our `evaluation_interval` to 5 seconds.

We did notice a bug in the metric you linked as well. Currently it is [taking the mean](https://github.com/camunda/zeebe/blob/6217d07ef93adfdffeb9426921cbdb66b94b6b75/monitor/grafana/zeebe.json#L1773) of the time range. This doesn't make sense, it's more accurate to change this to `lastNotNull`. 

Let me know if you have any more questions!
